logstash-core/src/test/java/org/logstash/FileLockFactoryMain.java

```diff
@@ -0,0 +1,23 @@
+package org.logstash;
+
+import java.io.IOException;
+
+/*
+ * This program is used to test the FileLockFactory in cross-process/JVM.
+ */
+public class FileLockFactoryMain {
+
+    public static void main(String[] args) {
+        // Timeout to ensure the process finishes.
+        long timeout = 20000;
+        try {
+            FileLockFactory.getDefault().obtainLock(args[0], args[1]);
+            System.out.println("File locked");
+            Thread.sleep(timeout);
```

Comment:
That's true, it will  reduce usability from the CLI. But I'm setting a timeout to avoid the CI system of creating zombie processes, for the case something goes wrong in JUnit side and this process isn't destroyed. Notice the way I spawn this process it is detached from the JUnit process.
Well, didn't think about manual tests when implementing this testcase. :)